hydromt_sfincs/workflows/flwdir.py

```diff
@@ -143,6 +144,11 @@ def river_source_points(
 
     # clip river to model gdf_mask
     gdf_riv = gdf_riv.to_crs(gdf_mask.crs).clip(gdf_mask.unary_union)
+    # keep only lines
+    gdf_riv = gdf_riv[
+        [t.endswith("LineString") for t in gdf_riv.geom_type]
+    ].reset_index(drop=True)
```

Comment:
Due to this reset_index, the tests were failing. I now fixed it on the test side, but we could also consider fixing it here. Points returned by the code were identical, only the riv_idx weren't.

Reply:
decided to remove riv_idx and test on the coordinates instead.